packages/salesforce/test/Adaptor.test.js

```diff
@@ -194,4 +195,86 @@ describe('Adaptor', () => {
       expect(result).to.eql('Nhamaonha 6a Classe 2023-10-09');
     });
   });
+
+  describe('query', () => {
+    it('should fetch all records', done => {
```

Comment:
I am struggling a bit with `fetch all records in a single page` because it doesn't say what is the maximum number of records per single page 🤔 . What about this `fetch up to 2000 records` ?

Reply:
I don't think it matters what the maximum number of records per page is.

What matters that if there is a another page, the function fetches it.

You've called that test `fetch up to 2000 records`, but I don't know why 2000 records is important. Nor does Hunter when this test starts failing on one of her feature branches.

The most important question here is: what are you actually trying to test?

Reply:
The reason 2000 doesn't matter is that this behaviour is controlled by the salesforce server. It sends `done` and `nextRecordsUrl` to tell you there is more data (it also sends the total count, which you could choose to use)

The adaptor doesn't say "if there are more than 2000 records, fetch another page". The adaptor says: if the request is not `done`, fetch another page.

Reply:
Are you trying to test that the function doesn't autofetch if there are no more items? Are you trying to test that the function can download 2000 items exactly per page? Are you trying to test that if there are many pages of data, the adaptor fetches them all?

Reply:
I want to test the function can download 2000 maximum rows but not necessary assert 2000 records because depending on your query you might get less than 2000 results.

Reply:
I still don't understand the value of that, but the way  I would usually test something like that is to test the bounds. Test with 2000 items returned by the server and assert on the correct behaviour, then test with 2001 items and assert on that. What happens if the server returns 2001 items anyway? (hint: nothing at all - the code doesn't care about this 2000 magic number). It would also be valuable to test with 0 items returned.

But because this 2000 limit is arbitrary, I don't think you need to go off and do that.

I'm closing this thread and starting a new one to move us forward.